js/autotheme.js

```diff
@@ -0,0 +1,174 @@
+/*global Color, ColorThief, Defer, EventDispatcher, MozActivity */
+
+'use strict';
+
+(function(exports) {
+  // is autotheming active
+  var active;
+
+  function loadImage(src) {
+    var defer = new Defer();
+
+    var img = new Image();
+    img.src = src;
+
+    if (img.complete) {
+      defer.resolve(img);
+    } else {
+      img.onload = () => defer.resolve(img);
+      img.onerror = defer.reject;
+    }
+
+    return defer.promise;
+  }
+
+  function blobFromURL(src) {
+    var defer = new Defer();
+
+    var xhr = new XMLHttpRequest();
+    xhr.addEventListener('load', () => {
+      defer.resolve(xhr.response);
+    });
+    xhr.addEventListener('error', defer.reject);
+
+    xhr.open('GET', src);
+    xhr.responseType = 'blob';
+    xhr.send();
+
+    return defer.promise;
+  }
+
+  var AutoTheme = {
```

Comment:
this is a singleto. ideally I'd like it to be instanceable so that I could have different instances for different themes. But I don't know, after all it works quite well here :)